### DISCUSSION TITLE
Fill pattern to image

### DISCUSSION THREAD

**Discussion by flappy-07** [ImageMagick/ImageMagick#3936] [Help]
Hi I am new to Imagemagick and would need help in filling pattern images to another image.

I want to add one skin pattern on the border and another on the inside. Can anyone please help.

Image 1: https://pasteboard.co/KbfG8qU.png
Skin 1: https://pasteboard.co/KbfGE4M.jpg
Skin 2: https://pasteboard.co/KbfGUy56.png

**Comment by snibgo** [comment]
What version of IM, on what platform?

> I want to add one skin pattern on the border and another on the inside.

Which pixels are "on the border"? Which pixels are "on the inside?"

**Reply by flappy-07** [reply]
IM version: 6.9 in windows. Image 1 is a transparent image created in paint 3d. Not sure about which pixels

**Comment by snibgo** [comment]
I don't understand what you mean by "on the border" and "on the inside". So I don't know where in "Image 1" you want the skin images to be tiled.

**Reply by flappy-07** [reply]
So this the image that i manually created just for reference: https://pasteboard.co/KbjoaSp.png
The black borders have now the skin1 image and the inside all borders we have the 2nd skin image. The gaps that is present in the image should be filled with 2nd skin image.

**Comment by snibgo** [comment]
Is this what you want? Windows BAT syntax:
```
magick ^
  input5.png +write mpr:INP ^
  -fuzz 90%% ^
  -fill blue -draw "color 0,0 floodfill" ^
  -channel A -negate +channel ^
  +write mpr:ALP0 ^
  -tile skin2.jpg -draw "color 0,0 reset" ^
  mpr:ALP0 ^
  -compose CopyOpacity -composite ^
  ( +clone ^
    -tile skin1.jpg -draw "color 0,0 reset" ^
    mpr:INP ^
    -compose CopyOpacity -composite ^
  ) ^
  +swap ^
  -compose Over -composite ^
  c.png
```
The output, c.png, is:
![c](https://user-images.githubusercontent.com/33812368/125869684-c3a43ec6-dc0b-4099-b329-b91303660f32.png)

**Reply by flappy-07** [reply]
thank you so much, this is what i needed
